components/function-controller/internal/controllers/serverless/function_controller.go

```diff
@@ -147,8 +147,16 @@ func (r *FunctionReconciler) onJobChange(ctx context.Context, log logr.Logger, i
 }
 
 func (r *FunctionReconciler) equalJobs(existing batchv1.Job, expected batchv1.Job) bool {
-	// Compare image argument
-	return existing.Spec.Template.Spec.Containers[0].Args[0] == expected.Spec.Template.Spec.Containers[0].Args[0]
+	// Compare arguments as it contains image tag
+	if len(existing.Spec.Template.Spec.Containers[0].Args) != len(expected.Spec.Template.Spec.Containers[0].Args) {
+		return false
+	}
+	for i, arg := range existing.Spec.Template.Spec.Containers[0].Args {
+		if expected.Spec.Template.Spec.Containers[0].Args[i] != arg {
+			return false
+		}
+	}
```

Comment:
isn't there a bug somewhere? what if those slices have the same elements but in different order

Reply:
It shouldn't be a problem as Job is created by the controller, but in fact, we should only compare `--destination` arg as it contains image tag. Other args shouldn't trigger build.

Reply:
So, I will change it.